Missing 2.4 changes + Menu updates

Doxorn opened the pull request:
All comments by me are _(in brackets and have italic font like this.)_

- Added **skillcalc.txt** (haven't changed anything in it but it was missing)

**Custom menu updates:**
- Character Balancing Menu title updated to D2R Reimagined Character Balancing
- Modinfo Menu title updated to D2R Reimagined Crafting Recipes
- Barbarian info panel updated with the new information _(blizzard 2.4 changes which were unmodified were omitted, including other class changes here)_

**2.4 update references:**
https://news.blizzard.com/en-us/diablo2/23788293/diablo-ii-resurrected-patch-2-4-ladder-now-live
Left original update notes intact, written notes in brackers

**AMAZON/ASSASSIN**
- no changes

**DRUID**
- BakeIt working on it #NotMyJob

**NECROMANCER**
Bone Spear:
- Teeth synergy increased from 7% to 8%
- Bone Wall synergy increased from 7% to 8%
- Bone Spear synergy increased from 7% to 8%
- Bone Prison synergy increased from 7% to 8%

Bone Spirit:
- Teeth synergy increased from 6% to 8%
- Bone Wall synergy increased from 6% to 8%
- Bone Spear synergy increased from 6% to 8%
- Bone Prison synergy increased from 6% to 8%

Weaken:
- Damage reduction now scales based on level, from just 33%, to now 33% and +1% per level
- Skill description updated _(overwritten D2R-R skilldesc with Blizzard 2.4 skilldesc, idk what changed)_
> Blizzard Dev Comments: For the “Weaken” skill, we added new damage reduction scaling to give more reason to spend additional points on the skill and to give it a more powerful identity.

Decrepify:
- Tooltip updated to now show the status changes

BARBARIAN:
Sword Mastery (Melee Mastery in D2R-R)
- Increased base attack rating bonus to 40% (from 28%)

Leap:
- Baseline minimum distance increased from 4 to 8 _(exact numbers weren't given in the patch notes but I checked in the .txt file so it's correct)_
- Speed of the leap motion increased by 75% _(reduced synegy scaling from 6% to 5% to match that)_
- Leap now has a minimum jump height and duration
- Tooltip updated to now display Knockback Radius

Leap Attack:
- Speed of the leap motion increased by 75% _(It was 100% pre-2.4 and 150% in Reimagined. Now it's 175%, reduced synegy scaling from 2% to 1% to match that)_
- Leap now has a minimum jump height and duration
- Attack Rating baseline increased from 50% to 100%
- Attack Rating per level increased from 15% to 20%
- The Barbarian now deals extra physical damage to surrounding enemies within a range of 4.6 yards when landing from a Leap Attack. Damage scales per level
- Leap synergy bonus applies to the AoE damage in addition to the attack _(added +10% Whirlwind to the synergy bonus as well)_
- Increased mana cost of Leap Attack from 9 to 10
- Leap Attack will be performed even when targeting adjacent enemies _(used to perform a basic attack)_
> Blizzard Dev Comments: Leap Attack has received a lot of changes: increased movement speed, smoother trajectory, improved controls, and increased attack rating and damage. Perhaps the biggest change is that Leap Attack now deals AOE damage to surrounding enemies when landing. This gives the Barbarian a strong multi-target skill as early as level 18. This damage does not scale with your weapon, so it's perfect for leveling up a Barbarian in Normal and Nightmare difficulties. The AOE damage won't make as much of an impact on Hell difficulty, but Leap Attack's primary attack and mobility will remain valuable.

**PALADIN**
Resist Fire:
- Tooltip now displays max resist bonus

Resist Cold:
- Tooltip now displays max resist bonus

Resist Lightning:
- Tooltip now displays max resist bonus

Blessed Aim:
- Tooltip now displays passive Attack Rating bonus

Holy Fire:
- Damage scales based on distance to the caster. Aura will deal double damage at melee range and linearly scale to normal damage at maximum range
- Tooltip: The maximum Area Damage values now incorporate the maximum proximity damage bonus
- Resist Fire Synergy increased from 18 to 21%
- Damage level scaling increased to 70%
- Salvation synergy increased from 6% to 10%

Holy Freeze:
- Damage scales based on distance to the caster. Aura will deal double damage at melee range and linearly scale to normal damage at maximum range
- Tooltip: The maximum Area Damage values now incorporate the maximum proximity damage bonus

Holy Shock:
- Damage scales based on distance to the caster. Aura will deal double damage at melee range and linearly scale to normal damage at maximum range
- Tooltip: The maximum Area Damage values now incorporate the maximum proximity damage bonus Sanctuary
- Damage scales based on distance to the caster. Aura will deal double damage at melee range and linearly scale to normal damage at maximum range
- Tooltip: The maximum Area Damage values now incorporate the maximum proximity damage bonus

Thorns:
- Now deals a flat amount of damage when attacked in addition to returning a percentage of damage taken to the enemy when hit
> Blizzard Dev Comments: We wanted to add more potential to the AoE (Area of Effect) Damage auras since they are a capable way for the Paladin to handle groups of monsters. By increasing damage based on distance to the caster, each AoE damage aura will have increased effectiveness while still maintaining the same cadence of visual VFX on the screen. Also, we saw that “Holy Fire” should scale better at higher skill levels to be consistent with general Fire Damage power compared to the other elements. For “Thorns”, we want to improve the gameplay around getting hit to deal damage, so we are making the skill deal flat damage based on attack attempts in addition to returning damage after getting hit. This will allow Thorns to have a more consistent combat performance instead of unpredictable and riskier results.

Sacrifice:
- Damage to self now reduces per level, from 8% to 8%-1%

Conversion:
- Maximum conversion chance increased from 50% to 90%

Smite:
- Tooltip now shows that the skill always hits

Holy Bolt:
- Now damages Demons in addition to Undead
- Removed synergy Blessed Hammer
- Damage increased by 50%
- Now pierces targets (Friendly and Hostile)
- Prayer synergy increased from 15% to 20%

Fist of the Heavens:
- Now damages Demons in addition to Undead
- Casting Delay reduced from 1 second to 0.4 seconds
- Casting Delay will no longer share its cooldown with other skills that have Casting Delays
- Tooltip updated to now display Casting Delay
- Holy Bolts now pierce targets
- No longer forces you to attack when out of mana
- Improved targeting functionality for auto acquiring enemies

**SORCERESS**
Frost Nova:
- Damage level scaling increased by about 25%

Blizzard:
- Casting Delay will no longer share its cooldown with other skills that have Casting Delays
- Tooltip updated to now display Casting Delay

Frozen Orb:
- Casting Delay will no longer share its cooldown with other skills that have Casting Delays
- Tooltip updated to now display Casting Delay

Frozen Armor:
- Base duration increased from 120 seconds to 144 seconds

Shiver Armor:
- Damage level scaling increased by about 25%
- Base duration increased from 120 seconds to 144 seconds

Chilling Armor:
- Damage level scaling increased by about 200%
- Duration bonus per level increased from 6 seconds to 12 seconds
- Chill length bonus per level for level 16+ increased from 1 second to 2 seconds (matching Shiver Armor)
- Defense baseline increased from 45% to 60%
- Defense bonus per level increased from 5% to 7%
- Frozen Armor damage synergy increased from 7% to 9%
- Chilling Armor damage synergy increased from 7% to 9%
> Blizzard Dev Comments: With the cold armor skills, we want to further diversify them by ensuring that each has distinctive advantages. After these changes, “Chilling Armor” will have the greatest Defense bonus and damage potential while still only triggering on ranged attacks. Also, “Frost Nova” is receiving damage scaling improvements to give it more power where it was lacking in the later game.

Nova:
- Nova mana cost reduced. The initial mana cost is now 13 from 15

Thunder Storm:
- Duration baseline increased from 32 seconds to 144 seconds
- Duration bonus per level increased from 8 seconds to 24 seconds Blizzard Dev Comments: We saw an opportunity to add new synergies for lesser used Lightning skills to increase their damage potential and create new skill build possibilities.

Inferno:
- Base damage and level scaling increased by about 75% (also to compensate for the incorrect damage calculation in the past)
- Warmth synergy increased from 13% to 16%
- Controls are updated so that the skill now casts in free form and no longer locks onto one target at a time
- Mana Cost baseline reduced from 7 Mana per second to 4 Mana per second
- Range baseline increased by 75%

Blaze:
- Damage scaling increased by about 60%

Fire Wall:
- Casting Delay will no longer share its cooldown with other skills that have Casting Delays
- Tooltip updated to now display Casting Delay

Meteor:
- Casting Delay will no longer share its cooldown with other skills that have Casting Delays
- Tooltip updated to now display Casting Delay

Hydra:
- You can now spawn multiple hydras on controller when holding the button down _(confirmation needed)_
- Casting delay removed
- Limited max number of Hydra groups to 6
> Blizzard Dev Comments: We wanted to improve the lesser used Fire skills and give more reasons to use them. Inferno is receiving player control improvements with free form casting, like Arctic Blast, and better damage scaling. Blaze is an often-ignored skill, so we wanted to free up synergy requirements and provide more utility and power with damage increases and a new Run/Walk Speed buff for extra utility. Lastly, with Hydra, we wanted to improve the gameplay of summoning Hydras to make it easier to use to skill in combat situations. The removal of Casting Delay and the new limitation on max Hydras will match the maximum Hydra count achievable from before the changes, but now players will have a much easier time to summon that amount.

Bakelt reviewed (commented):
Skills.txt
Frozen Orb - Reduce the delay back to 20 from 25. 
Holy Bolt - I'd revert all changes and keep what we have implemented
Fist of the Heavens - I'd revert all changes and keep what we have implemented.
Frenzy - Is this duration only? Just revert and do par7*2

Skilldesc.txt
Iron Maiden - This was completely blown out with Leap. Copy paste issue?
Holy Bolt - Revert. We have our own changes

Doxorn commented:
> Skills.txt Frozen Orb - Reduce the delay back to 20 from 25. Holy Bolt - I'd revert all changes and keep what we have implemented Fist of the Heavens - I'd revert all changes and keep what we have implemented. Frenzy - Is this duration only? Just revert and do par7*2
> 
> Skilldesc.txt Iron Maiden - This was completely blown out with Leap. Copy paste issue? Holy Bolt - Revert. We have our own changes

skills.txt
Frozen Orb local delay changed back from 25 to 20.
Holy Bolt - reverted all changes
Fist of the Heavens - reverted all changes
Frenzy - changed it to naturally get +0.2sec duration with each soft skill point

skilldesc.txt
Holy Bolt - reverted all changes
Fist of the Heavens - reverted all changes
Iron Maiden - is back

Bakelt reviewed (approved):
Requested changes have been made. Looks good!
